src/main/java/org/opentripplanner/routing/graphfinder/NearbyStop.java

```diff
@@ -81,7 +84,13 @@ public static NearbyStop nearbyStopForState(State state, StopLocation stop) {
         effectiveWalkDistance += edge.getEffectiveWalkDistance();
       }
       else if (edge instanceof PathwayEdge) {
-        effectiveWalkDistance += edge.getDistanceMeters();
+        int traversalTime = ((PathwayEdge) edge).getTraversalTime();
+        if (traversalTime > 0) {
+          distanceIndependentTime += traversalTime;
+        }
+        else {
+          effectiveWalkDistance += edge.getDistanceMeters();
+        }
```

Comment:
I see that we do not account for stairs and slope. The pathways have `max_slope`, but no slope - so it might not be possible to use the slope. Depending on how this is used I expect using the max_slop as slop and calculating the effectiveWalkDistance would be better than ignoring it.

 I would also love to see for all edges just:
```
effectiveWalkDistance += edge.getEffectiveWalkDistance();
distanceIndependentTime += edge.getDistanceIndependentTime();
```
Then override these to methods in every edge that has one or both of these. The default implementation should return zero(0).

You do not have to fix this, you pawed the way for fixing another problem here, so hopefully we can fix the "now missing" distanceIndependentTime in the street graph, and at the same time fix the `instanceof` problem. 

But, if you can use the `edge.getEffectiveWalkDistance()` instead of the `edge.getDistanceMeters()` for PathwayEdge that would be great. Also remove the if/else, it is not needed. If you want to add a check you can do it inside the  `getEffectiveWalkDistance()` method.

Reply:
:+1: I've updated `PathwayEdge` to override the methods and removed the special casing.